packages/edit-site/src/components/left-sidebar/navigation-panel/menus/template-parts.js

```diff
@@ -15,13 +15,10 @@ import TemplateNavigationItems from '../template-navigation-items';
 
 export default function TemplatePartsMenu( { onActivateItem } ) {
 	const templateParts = useSelect( ( select ) => {
-		const currentTheme = select( 'core' ).getCurrentTheme()?.textdomain;
```

Comment:
Note: this prevented demo template parts from showing up in the sidebar, so we'll need to find a better way to handle thhis.

Reply:
Why use the `textdomain` property of the theme? I was trying to find a semantic reason for preferring this property over `stylesheet`, but I don't see one. Looking at the use of `currentTheme` it seems to be used as the canonical theme identifier, which normally would be the theme's `stylesheet` property.

Practically speaking, `textdomain` isn't required to match `stylesheet` and — although the field is required by the Theme Repository — omitting it doesn't break the theme.

Also cc @Addison-Stavlo for the use of `textdomain` in #23254

Reply:
@mcsf Interesting, thanks for letting us know!  I thought of it as the identifier but did not realize it could be omitted.  I'l update these to use stylesheet.  I also noticed that global styles is using the `textdomain` field as well, but it may be for a different reason? https://github.com/WordPress/gutenberg/blob/270975190b5e539dc291be73749af3fb7e557c42/lib/global-styles.php#L144

Reply:
https://github.com/WordPress/gutenberg/pull/25995

Reply:
For context: IIRC, global styles used text domain because that was what FSE was using at the time + the thought was that it was the canonical identifier of the theme. In looking at this, I learned that it's not and that the `stylesheet` field that Miguel recommends using is what WordPress core uses instead when it wants a unique identifier ([example](https://github.com/WordPress/gutenberg/pull/26061)).

It looks like there are use cases in which the text-domain is not necessary (child themes that don't add any new string, themes not distributed through the WP repository) so any code that uses it as theme identifier can potentially break. Created a PR to fix `lib/global-styles.php` at https://github.com/WordPress/gutenberg/pull/26061

Would you follow suit and migrate FSE to use `stylesheet` as the new affix for the identifier? Do we need an issue to track this? I'm happy to help, but also have no experience with FSE so can miss tons of contexts as to what things may break by changing the identifier.